rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -611,6 +605,26 @@ public String getNodeMBeanHistory(@HeaderParam("sessionid") String sessionId,
         return rmProxy.getNodeMBeanHistory(nodeJmxUrl, objectName, attrs, range);
     }
 
+    @Override
+    @GET
+    @GZIP
+    @Produces("application/json")
+    @Path("nodes/mbean/history")
+    public String getNodesMBeanHistory(@HeaderParam("sessionid") String sessionId,
+            @QueryParam("nodejmxurl") Set<String> nodesJmxUrl, @QueryParam("objectname") String objectName,
+            @QueryParam("attrs") List<String> attrs, @QueryParam("range") String range)
+            throws InstanceNotFoundException, IntrospectionException, ReflectionException, IOException,
+            NotConnectedException, MalformedObjectNameException, NullPointerException, MBeanException {
+
+        // checking that still connected to the RM
+        RMProxyUserInterface rmProxy = checkAccess(sessionId);
+        Map nodesMBeanHistoryMap = new HashMap<String, String>();
+        for (String nodeJmxUrl : nodesJmxUrl) {
+            nodesMBeanHistoryMap.put(nodeJmxUrl, rmProxy.getNodeMBeanHistory(nodeJmxUrl, objectName, attrs, range));
+        }
+        return nodesMBeanHistoryMap.toString();
```

Comment:
@aminelouati if we return "nodesMBeanHistoryMap", it will call automatically toString() right? so maybe we don't need to put toString(). (I'm not sure though)

Reply:
Maybe better to return a JSON object. I am updating the PR. 
Thanks